protocol_connect.go

```diff
@@ -1151,10 +1151,15 @@ func (d *connectWireDetail) MarshalJSON() ([]byte, error) {
 	}
 	// Try to produce debug info, but expect failure when we don't have
 	// descriptors.
-	var codec protoJSONCodec
-	debug, err := codec.Marshal(d.pb)
-	if err == nil && len(debug) > 2 { // don't bother sending `{}`
-		wire.Debug = json.RawMessage(debug)
+	if wire.Value != "" { // don't bother sending `{}`
+		msg, err := d.pb.UnmarshalNew()
+		if err == nil {
+			var codec protoJSONCodec
+			debug, err := codec.Marshal(msg)
+			if err == nil {
+				wire.Debug = debug
+			}
+		}
```

Comment:
In the common case where the user created the error detail using `NewErrorDetail`, this feels a little wasteful - we had the original protobuf message in hand. Could we change the `ErrorDetail` struct to also keep a reference to the original protobuf message so we don't always need to go through an unmarshal here?

Reply:
Done.